### PR TITLE
add VMware provider to ManageIQ module

### PR DESCRIPTION
**Pull request by snecklifter**
##### SUMMARY
Currently the manageiq remote management module only supports
OpenShift, AWS and oVirt. This adds the ability to create VMware
providers by adding two additional parameters for configuring a
host VNC range.

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
remote_management/manageiq

##### ANSIBLE VERSION
```
ansible 2.5.0 (devel aa50487ed7) last updated 2017/12/20 21:08:51 (GMT +100)
```

##### ADDITIONAL INFORMATION
This has been tested using Red Hat OpenTLC LE

### CODE REVIEW

File: lib/ansible/modules/remote_management/manageiq/manageiq_provider.py [hunk 1]
```diff
@@ -46,6 +46,14 @@
     description: The provider region name to connect to (e.g. AWS region for Amazon).
     required: false
     default: null
+  host_default_vnc_port_start:
+    required: false
```
Comment (yaacov) [inline lib/ansible/modules/remote_management/manageiq/manageiq_provider.py:50]: you are missing the version_add here, like in:
https://github.com/ansible/ansible/blob/devel/lib/ansible/utils/module_docs_fragments/influxdb.py#L32

**Comment by yaacov** [comment]
shipit 🐑

**Comment by snecklifter** [comment]
@cben any chance of a review?

File: lib/ansible/modules/remote_management/manageiq/manageiq_provider.py [hunk 2]
```diff
@@ -37,7 +37,7 @@
   type:
     description: The provider's type.
     required: true
-    choices: ['Openshift', 'Amazon', 'oVirt']
+    choices: ['Openshift', 'Amazon', 'oVirt', 'VMware']
```
Comment (cben) [inline lib/ansible/modules/remote_management/manageiq/manageiq_provider.py:40]: Is there a way to document VMware is new in 2.5? (oVirt might be too BTW)

File: lib/ansible/modules/remote_management/manageiq/manageiq_provider.py [hunk 3]
```diff
@@ -46,6 +46,16 @@
     description: The provider region name to connect to (e.g. AWS region for Amazon).
     required: false
     default: null
+  host_default_vnc_port_start:
+    required: false
+    default: null
+    description: The first port in the host VNC range. defaults to None.
+    version_added: "2.5"
+  host_default_vnc_port_end:
+    required: false
```
Comment (cben) [inline lib/ansible/modules/remote_management/manageiq/manageiq_provider.py:55]: what happens if user specifies only one (e.g. end without start)?

File: lib/ansible/modules/remote_management/manageiq/manageiq_provider.py (hunk 2 above)
Comment (snecklifter) [reply lib/ansible/modules/remote_management/manageiq/manageiq_provider.py:40]: Not sure, I thought adding the example config was sufficient but let me know if there is some other way.

File: lib/ansible/modules/remote_management/manageiq/manageiq_provider.py (hunk 3 above)
Comment (snecklifter) [reply lib/ansible/modules/remote_management/manageiq/manageiq_provider.py:55]: yeah, i think a required_together should do it. Would you mind if I submitted that as a follow-up patch as I have removed my old repo this commit comes from after getting in a git mess.

Comment (snecklifter) [reply lib/ansible/modules/remote_management/manageiq/manageiq_provider.py:55]: Something like:

```
required_together = (['host_default_vnc_port_start', 'host_default_vnc_port_end'])
```

Comment (cben) [reply lib/ansible/modules/remote_management/manageiq/manageiq_provider.py:55]: sure, that's fine

**Comment by cben** [comment]
I don't like the increasing lists of positional parameters, have an idea how I want to refactor but it's out of scope for this PR.
LGTM.
shipit

**Pull request merged**